src/server/system_services/pool_controllers.js

```diff
@@ -117,6 +121,26 @@ class ManagedStatefulSetPoolController extends PoolController {
         }
     }
 
+    async upgrade(image) {
+        dbg.log0(`ManagedStatefulSetPoolController::upgrade: upgrading (if needed) stateful set ${
+            this._statefulset_name
+        } to use ${
+            image
+        } image`);
+
+        await kube_utils.patch_resource(
+            'StatefulSet',
+            this._statefulset_name,
+            { spec: { template: { spec: { containers: [{ name: 'noobaa-agent', image }] } } } }
+        );
+
+        const pods = await this._list_pods();
+        await Promise.all(pods.map(async pod => {
+            const pod_name = pod.metadata.name;
+            return kube_utils.wait_for_condition('pod', pod_name, 'ready', 15 * 60 * 1000);
+        }));
```

Comment:
do we really need to wait for the agents to be ready? this can block the entire server from starting in case there is a problem in one of the agents

Reply:
never mind. I see now that pool_server init does not wait